test/e2e/tests/dapp-interactions.spec.js

```diff
@@ -0,0 +1,125 @@
+const { strict: assert } = require('assert');
+const {
+  convertToHexValue,
+  withFixtures,
+  connectDappWithExtensionPopup,
+} = require('../helpers');
+
+describe('Dapp interactions', function () {
+  let windowHandles;
+  let extension;
+  let popup;
+  const ganacheOptions = {
+    accounts: [
+      {
+        secretKey:
+          '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+        balance: convertToHexValue(25000000000000000000),
+      },
+    ],
+  };
+  it('should a notification be triggered despite Metamask being locked', async function () {
+    await withFixtures(
+      {
+        dapp: 1,
+        fixtures: 'imported-account',
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        // Connect to Dapp0
+        await connectDappWithExtensionPopup(driver, 0);
+        windowHandles = await driver.getAllWindowHandles();
+        extension = windowHandles[0];
+
+        // Lock Account
+        await driver.switchToWindow(extension);
+        await driver.clickElement('.account-menu__icon');
+        await driver.clickElement({ text: 'Lock', tag: 'button' });
+
+        // Trigger Notification
+        await driver.switchToWindowWithTitle('E2E Test Dapp', windowHandles);
+        await driver.clickElement('#addEthereumChain');
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+        const notification = await driver.findElement({
+          text: 'Allow this site to add a network?',
+          tag: 'h3',
+        });
+
+        assert.ok(notification, 'Dapp action does not appear in Metamask');
```

Comment:
Should this be `'Dapp action _does_ appear in Metamask'`? This should verify that after unlocking, the user does actually see the add ethereum chain notification page.

Reply:
ah, okay, that makes sense. Thanks for explaining